src/shared/containers/ImagePreviewContainer.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { notification } from 'antd';
+import { useAsyncEffect } from 'use-async-effect/';
+import { useNexusContext } from '@bbp/react-nexus';
+import { Resource, NexusFile } from '@bbp/nexus-sdk';
+
+import { isFile } from '../utils/nexusMaybe';
+import { getResourceLabelsAndIdsFromSelf } from '../utils';
+import ImagePreviewComponent from '../components/Images/Preview';
+
+// Only preview images lower than 3MB in size
+const MAX_BYTES_TO_PREVIEW = 3000000;
+
+const ImagePreviewContainer: React.FunctionComponent<{
+  resource: Resource;
+  maxBytes?: number;
+}> = props => {
+  const nexus = useNexusContext();
+  const { resource, maxBytes = MAX_BYTES_TO_PREVIEW } = props;
+  const {
+    orgLabel,
+    projectLabel,
+    resourceId,
+  } = getResourceLabelsAndIdsFromSelf(resource['_self']);
+
+  const [{ busy, imageSrc, error }, setImage] = React.useState<{
+    busy: boolean;
+    error: Error | null;
+    imageSrc: string | null;
+  }>({
+    busy: false,
+    imageSrc: null,
+    error: null,
+  });
+
+  const makePreviewImage = (imageSrc: string) => {
+    const img = new Image();
+    img.src = imageSrc;
+    return <ImagePreviewComponent busy={busy} image={img} />;
+  };
+
+  useAsyncEffect(
+    async isMounted => {
+      if (!isMounted()) {
```

Comment:
can it be `!isMounted() || !isFile(resource)`?

Reply:
good idea